src/core/public/chrome/ui/header/header.tsx

```diff
@@ -191,6 +191,7 @@ export class Header extends Component<HeaderProps, State> {
 
     const className = classnames(
       'chrHeaderWrapper',
+      'disableCoreNavigation',
```

Comment:
So, I started by removing the `onClick` attribute of the navlinks

https://github.com/elastic/kibana/blob/c08954922afcdefa2614e6d4e5de416025cc10dd/src/core/public/chrome/ui/header/nav_link.tsx#L67-L78

However there is a behavior difference between our global handler and the navlink's: When on a given app, the associated navlink will/should perform a SPA navigation. Using our global handler would perform a full page refresh, as it will not intercept clicks directing to the currently active app. So I disabled our handler for the header instead.